qiskit/quantum_info/operators/operator.py

```diff
@@ -556,20 +559,42 @@ def power(self, n: float) -> Operator:
         if self.input_dims() != self.output_dims():
             raise QiskitError("Can only power with input_dims = output_dims.")
         ret = _copy.copy(self)
-        if isinstance(n, int):
-            ret._data = np.linalg.matrix_power(self.data, n)
+        if isinstance(exponent, int):
+            ret._data = np.linalg.matrix_power(self.data, exponent)
         else:
             import scipy.linalg
 
             # Experimentally, for fractional powers this seems to be 3x faster than
-            # calling scipy.linalg.fractional_matrix_power(self.data, n)
+            # calling scipy.linalg.fractional_matrix_power(self.data, exponent)
             decomposition, unitary = scipy.linalg.schur(self.data, output="complex")
             decomposition_diagonal = decomposition.diagonal()
-            decomposition_power = [pow(element, n) for element in decomposition_diagonal]
+            decomposition_power = [pow(element, exponent) for element in decomposition_diagonal]
             unitary_power = unitary @ np.diag(decomposition_power) @ unitary.conj().T
             ret._data = unitary_power
```

Comment:
I just made #13358 to sort out the SciPy 1.14 issues, which as a side-effect modified all this code to use the `fraction_matrix_power` form with a little rotation to push the branch cut somewhere less likely to cause issues (Lev's idea).  That obsoletes the "bugfix" here in `power` - sorry, I was just trying to get SciPy 1.14 working quicker.

That said, I think the new method / `assume_unitary` keyword arg to this function is still a very useful idea, so this note is just to make sure that we apply the same branch-cut protection trick here too, and ideally add some tests that it's stable for various roots.